src/lib/spinel/spinel.h

```diff
@@ -4937,11 +4941,17 @@ typedef uint32_t spinel_prop_key_t;
 
 #define SPINEL_HEADER_IID_SHIFT 4
 #define SPINEL_HEADER_IID_MASK (3 << SPINEL_HEADER_IID_SHIFT)
-
-#define SPINEL_HEADER_IID_0 (0 << SPINEL_HEADER_IID_SHIFT)
-#define SPINEL_HEADER_IID_1 (1 << SPINEL_HEADER_IID_SHIFT)
-#define SPINEL_HEADER_IID_2 (2 << SPINEL_HEADER_IID_SHIFT)
-#define SPINEL_HEADER_IID_3 (3 << SPINEL_HEADER_IID_SHIFT)
+#define SPINEL_HEADER_IID(iid) (static_cast<uint8_t>((iid) << SPINEL_HEADER_IID_SHIFT))
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
```

Comment:
I don't think it is a good idea to use the macro `OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE` in this file. This file doesn't include any configuration header files. The configuration of `OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE` in the configuration header file may not affects this file.

Reply:
Good catch; modified the code to remove openthread config from this file.